proto/warehouse/warehouse.proto

```diff
@@ -109,11 +113,69 @@ message RetryWHUploadsResponse {
 }
 
 message ValidateObjectStorageRequest {
-  string type=1;
-  google.protobuf.Struct config=2;
+  string type = 1;
+  google.protobuf.Struct config = 2;
 }
 
 message ValidateObjectStorageResponse{
-  bool isValid=2;
-  string error=3;
+  bool isValid = 2;
+  string error = 3;
+}
+
+message RetrieveFailedBatchesRequest{
+  string workspaceID = 1;
+  string destinationID = 2;
+  string intervalInHours = 3;
+}
+
+message RetrieveFailedBatchesResponse{
+  repeated FailedBatchInfo infos = 1;
+}
+
+message FailedBatchInfo{
+  string errorCategory = 1;
+  string sourceID = 2;
+  int64 count = 3;
```

Comment:
Does this represent the number of events failed in this batch? If yes, can we rename it to `failedEventsCount`?